action/protocol/staking/vote_bucket.go

```diff
@@ -213,8 +217,13 @@ func bucketKey(index uint64) []byte {
 	return append(key, byteutil.Uint64ToBytesBigEndian(index)...)
 }
 
-func calculateVoteWeight(c genesis.VoteWeightCalConsts, v *VoteBucket, selfStake bool) *big.Int {
+// CalculateVoteWeight calculates the vote weight
+func CalculateVoteWeight(c genesis.VoteWeightCalConsts, v *VoteBucket, selfStake bool) *big.Int {
 	remainingTime := v.StakedDuration.Seconds()
+	if !v.isNative() {
+		// according to produce one block per 5 seconds
+		remainingTime = float64(v.StakedDurationBlockNumber) * 5
```

Comment:
don't hard code constant

Reply:
yes, by maintain the `StakedDuration` field also for the nft bucket, we can remove the additional case here.